internal/mode/static/nginx/runtime/verify.go

```diff
@@ -76,17 +77,13 @@ func (c *verifyClient) WaitForCorrectVersion(ctx context.Context, expectedVersio
 		25*time.Millisecond,
 		true, /* poll immediately */
 		func(ctx context.Context) (bool, error) {
-			version, err := c.GetConfigVersion()
-			if err != nil {
-				return false, err
-			}
-			if version == expectedVersion {
-				return true, nil
-			}
-			return false, nil
+			version, err := c.getConfigVersion()
+			return version == expectedVersion, err
 		}); err != nil {
-		return fmt.Errorf("could not get expected version %v: %w", expectedVersion, err)
+		if errors.Is(err, context.DeadlineExceeded) {
+			err = fmt.Errorf("config version check didn't return expected version within the deadline")
```

Comment:
Should we add the expected version to this error message?

Reply:
It's added below, so the error returned would be:
`could not get expected config version 3: config version check didn't return expected version within the deadline`
but I'll add it here as well to be super clear about it